runtime/vm/createramclass.cpp

```diff
@@ -1633,53 +1643,87 @@ loadSuperClassAndInterfaces(J9VMThread *vmThread, J9ClassLoader *classLoader, J9
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 /**
  * Attempts to pre-load classes of fields with Q signatures and no static
- * access modifier set.
+ * access modifier set. Also records largest field type in largestFieldType.
+ * 0 for single 1 for ref and 2 for double. Records total number of
+ * flattenable instance fields in flattenableInstanceFieldCount.
  *
  * Caller should not hold the classTableMutex.
  *
  * Return TRUE on success. On failure, returns FALSE and sets the
  * appropriate Java error on the VM.
  */
 static BOOLEAN
-loadFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *classLoader, J9ROMClass *romClass, UDATA classPreloadFlags, UDATA packageID, J9Module *module)
+loadFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *classLoader, J9ROMClass *romClass, UDATA classPreloadFlags, UDATA packageID, J9Module *module, UDATA *largestFieldType, J9FlattenedClassCache *flattenedClassCache)
 {
 	J9ROMFieldWalkState fieldWalkState = {0};
 	J9ROMFieldShape *field = romFieldsStartDo(romClass, &fieldWalkState);
 	BOOLEAN result = TRUE;
+	UDATA flattenableInstanceFieldCount = 0;
+
 	/* iterate over fields and load classes of fields marked as QTypes */
 	while (NULL != field) {
 		const U_32 modifiers = field->modifiers;
-		J9UTF8 *signature = J9ROMFIELDSHAPE_SIGNATURE(field);
-		U_8 *signatureChars = J9UTF8_DATA(signature);
-		if (('Q' == signatureChars[0]) && J9_ARE_NO_BITS_SET(modifiers, J9AccStatic)) {
-			J9Class *valueClass = internalFindClassUTF8(currentThread, signatureChars + 1, J9UTF8_LENGTH(signature) - 2, classLoader, classPreloadFlags);
-			if (NULL == valueClass) {
-				result = FALSE;
-				goto done;
-			} else {
-				J9ROMClass *valueROMClass = valueClass->romClass;
+		if (J9_ARE_NO_BITS_SET(modifiers, J9AccStatic)) {
+			J9UTF8 *signature = J9ROMFIELDSHAPE_SIGNATURE(field);
+			U_8 *signatureChars = J9UTF8_DATA(signature);
+			switch (signatureChars[0]) {
+			case 'Q':
+				{
```

Comment:
I'd just as soon not see another level of indent here. I'd either treat the case like an if (my preference), or move the curlies to the same indent as the case.